api_testing/apis/test_plan_api.py

```diff
@@ -16,8 +16,51 @@ def get_all_test_plans(self, **kwargs):
         self.info('Status code: {}'.format(response.status_code))
         return response
 
-    def get_completed_testplans(self):
-        response = self.get_all_test_plans()
+    def get_completed_testplans(self, **kwargs):
+        response = self.get_all_test_plans(**kwargs)
         all_test_plans = response.json()['testPlans']
         completed_test_plans = [test_plan for test_plan in all_test_plans if test_plan['status'] == 'Completed']
         return completed_test_plans
+    
+    def get_inprogress_testplans(self, **kwargs):
+        response = self.get_all_test_plans(**kwargs)
+        all_test_plans = response.json()['testPlans']
+        inprogress_test_plans = [test_plan for test_plan in all_test_plans if test_plan['status'] == 'InProgress']
+        return inprogress_test_plans
+
+    def get_testunits_in_testplan(self, id):
+        api = '{}{}/get/{}'.format(self.url, self.END_POINTS['test_plan_api']['list_all_test_plans'], id)
+        self.info('GET : {}'.format(api))
+        response = self.session.get(api, headers=self.headers, verify=False)
+        self.info('Status code: {}'.format(response.status_code))
+        testplan_data = response.json()['testPlan']
+        testunits_data = testplan_data['specifications']
+        return testunits_data
+
+    def get_testplan_with_quicksearch(self, quickSearchText, **kwargs):
+        api = '{}{}'.format(self.url, self.END_POINTS['test_plan_api']['list_all_test_plans'])
+        _payload = {"sort_value": "number",
+                    "limit": 20,
+                    "start": 0,
+                    "sort_order": "DESC",
+                    "filter": '{"quickSearch":"'+ quickSearchText + '","columns":["number","name"]}',
+                    "deleted": "0"}
+        payload = self.update_payload(_payload, **kwargs)
+        self.info('GET : {}'.format(api))
+        response = self.session.get(api, params=payload, headers=self.headers, verify=False)
+        self.info('Status code: {}'.format(response.status_code))
+        return response.json()['testPlans']
+
+    def get_testplan_with_filter(self, filter_option, filter_text, **kwargs):
+        api = '{}{}'.format(self.url, self.END_POINTS['test_plan_api']['list_all_test_plans'])
+        _payload = {"sort_value": "number",
+                    "limit": 20,
+                    "start": 0,
+                    "sort_order": "DESC",
+                    "filter": '{"' + filter_option + '":"' + filter_text + '","columns":["number","name"]}',
+                    "deleted": "0"}
+        payload = self.update_payload(_payload, **kwargs)
+        self.info('GET : {}'.format(api))
+        response = self.session.get(api, params=payload, headers=self.headers, verify=False)
+        self.info('Status code: {}'.format(response.status_code))
+        return response.json()['testPlans']
```

Comment:
re-implement it like the previous method

Reply:
done